ethereumj-core/src/main/java/org/ethereum/casper/core/CasperTransactionExecutor.java

```diff
@@ -0,0 +1,145 @@
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import static org.ethereum.util.BIUtil.toBI;
+
+public class CasperTransactionExecutor extends CommonTransactionExecutor {
+
+    public CasperTransactionExecutor(Transaction tx, byte[] coinbase, Repository track, BlockStore blockStore,
+                                     ProgramInvokeFactory programInvokeFactory, Block currentBlock) {
+
+        super(tx, coinbase, track, blockStore, programInvokeFactory, currentBlock, new EthereumListenerAdapter(), 0);
+    }
+
+    public CasperTransactionExecutor(Transaction tx, byte[] coinbase, Repository track, BlockStore blockStore,
+                                     ProgramInvokeFactory programInvokeFactory, Block currentBlock,
+                                     EthereumListener listener, long gasUsedInTheBlock) {
+
+        super(tx, coinbase, track, blockStore, programInvokeFactory, currentBlock, listener, gasUsedInTheBlock);
+    }
+
+    @Override
+    public void init() {
+        super.init();
+        // Already failed on common validation
+        if (execError != null) {
+            return;
+        }
+
+        // More validations for Casper
+
+        // EIP 208
+        if (Arrays.equals(tx.getSender(), Transaction.NULL_SENDER) && toBI(tx.getNonce()).compareTo(BigInteger.ZERO) > 0) {
+            execError(String.format("Null sender transaction should use 0 nonce, %s instead", toBI(tx.getNonce())));
+
+            return;
+        }
+
+        // EIP 208
+        if (Arrays.equals(tx.getSender(), Transaction.NULL_SENDER) &&
+                (toBI(tx.getValue()).compareTo(BigInteger.ZERO) > 0 || toBI(tx.getGasPrice()).compareTo(BigInteger.ZERO) > 0)) {
+            execError(String.format("Null sender transaction should have 0 value (actual %s), " +
+                    "and 0 gasprice (actual: %s)", toBI(tx.getValue()), toBI(tx.getGasPrice())));
+
+            return;
+        }
+    }
+
+    @Override
+    protected boolean isSignatureValid() {
+        return isCasperVote() || super.isSignatureValid();
+    }
+
+    @Override
+    public void execute() {
+
+        if (!readyToExecute) return;
+
+        if (!localCall && !isCasperVote()) {
+            track.increaseNonce(tx.getSender());
+
+            BigInteger txGasLimit = toBI(tx.getGasLimit());
+            BigInteger txGasCost = toBI(tx.getGasPrice()).multiply(txGasLimit);
+            track.addBalance(tx.getSender(), txGasCost.negate());
+
+            if (logger.isInfoEnabled())
+                logger.info("Paying: txGasCost: [{}], gasPrice: [{}], gasLimit: [{}]", txGasCost, toBI(tx.getGasPrice()), txGasLimit);
+        }
+
+        if (tx.isContractCreation()) {
+            create();
+        } else {
+            call();
+        }
+    }
+
+    private boolean isCasperVote() {
+        return isCasperVote(tx, ((CasperProperties) config).getCasperAddress());
+    }
+
+    public static boolean isCasperVote(Transaction transaction, byte[] casperAddress) {
```

Comment:
I think `CasperFacade` better fits this logic

Reply:
Agree, got this idea yesterday but didn't implement yet.